### PR TITLE
token-urls

### PR DESCRIPTION
**Pull request by tombola**
Adds tokenised urls, described in README.

### CODE REVIEW

**Comment by tombola** [comment]
I added and then removed `--recursive` option for isort, it is apparently not needed any more, I must have had old version of isort installed.

Also looks like my vscode was not using isort for python import sorts for some reason, so conflicted with tests, ran isort manually from cli should now pass....

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,7 +39,7 @@ You can format code and imports by calling:
 
 ```
 black wagtailsharing
-isort --recursive wagtailsharing
+isort wagtailsharing
```
Comment (zerolab) [inline CONTRIBUTING.md:42]: isort 5.x does recursive by default - https://timothycrosley.github.io/isort/docs/upgrade_guides/5.0.0/

File: wagtailsharing/views.py [hunk 2]
```diff
@@ -99,3 +107,27 @@ def serve(page, request, args, kwargs):
                 return result
 
         return response
+
+
+class TokenServeView(ServeView):
+    def dispatch(self, request, path):
+        sharing_site = self.get_sharing_site(request, path)
```
Comment (zerolab) [inline wagtailsharing/views.py:117]: should only react to GET requests

File: README.rst [hunk 3]
```diff
@@ -117,6 +117,12 @@ Shared pages will also have a new dropdown menu option that links to this sharin
 .. image:: https://raw.githubusercontent.com/cfpb/wagtail-sharing/master/docs/images/dropdown.png
     :alt: Dropdown with sharing link
 
+To use tokens in place of the page path on the sharing site, add the following setting:
+
+.. code-block:: python
+
+  WAGTAILSHARING_TOKENIZE_URL = True
+
```
Comment (zerolab) [inline README.rst:125]: worth adding a note that in some environments such as Heroku, `SERVER_PORT` set at release time and the real port is fowarded, so `USE_X_FORWARDED_PORT = True` is needed in settings

**Review by zerolab — CHANGES_REQUESTED** [review]

File: setup.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 install_requires = [
     "wagtail>=2.3,<2.10",
```
Comment (zerolab) [inline setup.py]:
```suggestion
    "wagtail>=2.3,<2.11",
```

since we're running on 2.10rc1

**Comment by zerolab** [comment]
Nice one. Why not open a WIP PR against the main repo to get any further feedback. Will see if I can provide tests

**Comment by zerolab** [comment]
<s>@tombola would be great if you could submit this as a PR against the main repo</s>
I take it back, just saw https://github.com/cfpb/wagtail-sharing/pull/38

**Pull request closed without merging**